### PR TITLE
Refactor To-Do List Server for Better Task Management

### PR DESCRIPTION
**Pull request by KhanyiCodes**
This pull request refactors the To-Do List server code to improve the structure, performance, and maintainability of the task management functionality. The changes are aimed at enhancing the way tasks are handled in the backend, including making the code more modular, improving error handling, validating inputs, and optimizing data storage

Key Changes:
Improved Code Structure and Modularity:
The server code has been refactored to follow better modular practices.
Before: The server file contained all the logic, which made it bulky and difficult to scale.
After: The logic has been organized into separate modules for clarity and maintainability.

Enhanced Task Management API:
Task Validation: We now ensure that all tasks being added to the list contain valid input. Tasks must have a non-empty text field before being added.
Completion Handling: Added validation to mark tasks as completed only when they exist. If the task doesn't exist, a 404 Not Found error is returned.
Error Handling: Proper error responses have been added. For instance, if a user tries to add a task with invalid input, they now receive a clear error message (400 Bad Request).

### CODE REVIEW

File: public/index.html
Comment (PhodzoW0l7) [inline public/index.html:1]: Well structured format.

File: public/script.js
Comment (PhodzoW0l7) [inline public/script.js:1]: Well structured format but I would suggest adding a message for creation and deletion of a task,

**Review by PhodzoW0l7 — COMMENTED** [review]
Well structured project.

**Review by bokangkhumalo — CHANGES_REQUESTED** [review]
Hi @KhanyiCodes,

🚀 Code Review Summary
Fantastic job on building the core functionality! Here are a few suggestions that could enhance readability and maintainability:

✍️ 1. Add Comments to Key Sections
The code currently lacks comments, which makes it challenging to follow, especially in complex areas like data processing and logic flow. Adding brief comments to summarize each section would make the code easier to read and understand.

🔍 2. Improve Naming for Clarity
Some function and variable names could be more descriptive to reflect their purpose. Clearer naming can often replace the need for comments, making the code more self-explanatory.

📄 3. Add Documentation or a README
Including a brief README or module-level docstrings to explain the overall structure and purpose of each module would be helpful for future contributors. This would give new readers a clear starting point.

Looking forward to seeing these changes

File: public/style.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.todo-item input[type="checkbox"] {
+    margin-right: 10px;
+}
+
+.todo-item.completed {
+    text-decoration: line-through;
+    color: #aaa;
+}
```
Comment (Tepanyega) [inline public/style.css]: hi @KhanyiCodes,
Can you ensure that when you mark your task as completed, the delete button is not displayed with a line-through?

**Review by Tepanyega — CHANGES_REQUESTED** [review]
Kindly ensure that the requested changes are made before the end of the day.

File: public/style.css (hunk 1 above)
Comment (KhanyiCodes) [reply public/style.css]: Hello @Tepanyega , yes I fixed it and placed it as a separate list without a line through the deleted button

File: public/script.js
Comment (KhanyiCodes) [reply public/script.js:1]: Thank you, I created pop up messages. 
Delete:
image
Create:
image

**Review by JKRakuba — APPROVED** [review]

**Review by KhanyiCodes — COMMENTED** [review]
New updates:
1. Edit the neon colours, layout structure, and make it for functional.
2. Added more text and paragraph, to explain to user.

**Review by PhodzoW0l7 — APPROVED** [review]
Everything looks okay and working.

**Review by Tepanyega — APPROVED** [review]
happy coding

File: public/app.js
Comment (alphineandre) [inline public/app.js:1]: Functionalities look good.

File: public/index.html
Comment (alphineandre) [inline public/index.html:1]: Structure looks good. But separate the task categories to account for access space around the list.

File: public/style.css
Comment (alphineandre) [inline public/style.css:1]: Design looks good. Nice touch with the neon.

**Review by alphineandre — APPROVED** [review]
Great Work!! Looks amazing...

**Review by Tepanyega — APPROVED** [review]

**Review by bokangkhumalo — APPROVED** [review]
Hi  @KhanyiCodes, Thanks for creating a Fully functional To-do list.

**Pull request merged**